kubeflow/trainer/api/trainer_client.py

```
@@ -136,6 +145,25 @@ def train(
+                )
+                return experimental_backend.train(
+                    runtime=runtime,
+                    initializer=initializer,
+                    trainer=trainer,
+                    options=options,
+                )
+            else:
+                raise ValueError(
+                    f"TransformersTrainer requires KubernetesBackend, "
+                    f"got {type(self.backend).__name__}"
+                )
```

Ruff (0.14.4)

Prefer `TypeError` exception for invalid type

(TRY004)